### ISSUE TITLE
Unify broker/gateway error reporting

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#1879]
After #1862, there is obviously a lot of overlap between the different error messages that we report. Most rejections of type `NOT_FOUND` obviously have very similar messages, most of type `INVALID_ARGUMENT` as well, etc. This also applies to normal errors, as we always try to follow the same pattern of `Expected to do X with context Y, but got Z`.

The goal here is to come up with an API to DRY up our code a little and make it easier for us and contributors to have consistent error messages.

Ideally it should then be easy to generate consistent error messages in our tests, which is particularly interesting when testing command rejection reasons without having brittle tests.

**Comment by npepinpe** [comment]
For command rejections, one of the first step would probably be to migrate all command processors to `CommandProcessor` and not some, and others simply `TypedRecordProcessor`, to ease providing a single entry point to write rejections through the `CommandControl`.

**Comment by npepinpe** [comment]
This was already done.
